### PR TITLE
Oe/86bz12tn4/realization stepper v1

### PR DESCRIPTION
**Pull request by guardian-001**
I ran this PR just to ensure that the number of files changed is not high. The work is not complete
folders and files structure will be updated in the last pr

### CODE REVIEW

**Review by Ghazi-Chaftar — COMMENTED** [review]
left some comments

File: src/api/architect/project/use-needs.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { client } from '@/api/common';
+import type { Need } from '@/types';
+
+type Response = Need[];
+type Variables = { id: number };
+
+export const useNeedsApi = createQuery<Response, Variables, AxiosError>({
+  queryKey: ['categories'],
+  fetcher: ({ id }) => {
+    // const [_key, { id }] = queryKey; // Destructure to get the id from queryKey
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/api/architect/project/use-needs.tsx:13]: done

File: src/api/architect/project/use-needs.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import type { AxiosError } from 'axios';
+import { createQuery } from 'react-query-kit';
+
+import { client } from '@/api/common';
+import type { Need } from '@/types';
+
+type Response = Need[];
+type Variables = { id: number };
+
+export const useNeedsApi = createQuery<Response, Variables, AxiosError>({
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/api/architect/project/use-needs.tsx:10]: Megdich tells me to name it like that

File: src/app/(supplier)/(public)/login.tsx [hunk 3]
```diff
@@ -13,11 +12,14 @@ export type LoginFormSupplierProps = {
 };
 export default function Login() {
   const router = useRouter();
-  const signIn = useAuth.use.signIn();
+  // const signIn = useAuth.use.signIn();
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/app/(supplier)/(public)/login.tsx:15]: done

File: src/app/(supplier)/(public)/login.tsx [hunk 4]
```diff
@@ -13,11 +12,14 @@ export type LoginFormSupplierProps = {
...
   useSoftKeyboardEffect();
 
   const onSubmit: LoginFormSupplierProps['onSubmit'] = () => {
-    signIn({ access: 'access-token', refresh: 'refresh-token' });
+    // signIn({
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/app/(supplier)/(public)/login.tsx:19]: done

File: src/modules/architect/realization/steps/approved-services.tsx [hunk 5]
```diff
@@ -1,32 +1,36 @@
...
     handleSubmit,
     // control,
-    // errors
+    errors,
   } = useCustomForm(ProjectRealizationSchema, {
     needs: formData.needs,
   });
-
+  // const { data, isPending, isError } = useNeedsApi({id:});
+  console.log(formData);
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/modules/architect/realization/steps/approved-services.tsx:25]: done

File: src/modules/architect/realization/steps/approved-services.tsx [hunk 6]
```diff
@@ -40,7 +44,23 @@ export default function ApprovedServices() {
           className="max-w-xs text-center text-sm text-description"
         />
       </View>
-
+      <View className="gap-4">
+        {/* {NeedsData.map((cardData, index) => (
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/modules/architect/realization/steps/approved-services.tsx:48]: done

File: src/modules/architect/realization/steps/project-details.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  );
+  const onSubmit = (data: DetailsFormType) => {
+    console.log(data);
+    setFormData((prev: any) => ({
+      ...prev,
+      ...data,
+    }));
+    onHandleNext();
+  };
+  // const projectNameError = errors.description?.message as TxKeyPath | undefined;
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/modules/architect/realization/steps/project-details.tsx:44]: done

File: src/modules/login/login-form.tsx [hunk 8]
```diff
@@ -31,7 +31,11 @@ export const LoginForm = () => {
   const onSubmit = (data: LoginFormType) => {
     login.mutate(data, {
       onSuccess: (response) => {
-        signIn({ access: response.access, refresh: response.refresh });
+        console.log(response);
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/modules/login/login-form.tsx:34]: done

**Pull request merged**